src/test/java/io/managed/services/test/quickstarts/steps/SpringBootSteps.java

```diff
@@ -0,0 +1,144 @@
+        });
+        builder.directory(Paths.get(quickstartRoot).toFile());
+        builder.redirectErrorStream(true); //merge input and error streams
+
+        Process process = builder.start();
+        BufferedReader stdInput = new BufferedReader(new InputStreamReader(process.getInputStream()));
+        while ((stdOutput = stdInput.readLine()) != null) {
+            System.out.println(stdOutput);
+        }
+    }
+
+    @When("you set the Kafka instance bootstrap server endpoint, service account credentials, and OAUTHBEARER token endpoint as environment variables for Spring app")
+    public void youSetTheKafkaInstanceBootstrapServerEndpointServiceAccountCredentialsAndOAUTHBEARERTokenEndpointAsEnvironmentVariablesForSpringApp() {
+
+        log.info("setting up map of all environment variables for python application");
+        envsMap.put("KAFKA_HOST", kafkaInstanceContext.requireKafkaInstance().getBootstrapServerHost());
+        envsMap.put("RHOAS_SERVICE_ACCOUNT_CLIENT_ID", serviceAccountContext.requireServiceAccount().getClientId());
+        envsMap.put("RHOAS_SERVICE_ACCOUNT_CLIENT_SECRET",
+                serviceAccountContext.requireServiceAccount().getClientSecret());
+        envsMap.put("RHOAS_SERVICE_ACCOUNT_OAUTH_TOKEN_URL", String.format("%s/auth/realms/%s/protocol/openid-connect/token", Environment.REDHAT_SSO_URI,
+                Environment.REDHAT_SSO_REALM));
+        log.debug(envsMap);
+    }
+
+    @When("you run Springboot client example consumer")
+    public void youRunSpringbootClientExampleConsumer() throws IOException {
+
+        log.info("package run Springboot client example consumer application");
+
+        // set path to root of quickstart
+        String quickstartRoot = this.repository.getAbsolutePath() + "/kafka-producer-spring";
+
+        // mvn compile exec:java -Dexec.mainClass="com.example.kafkademo.KafkaProducerExample"
+        ProcessBuilder builder = new ProcessBuilder("mvn", "compile", "exec:java", "-Pconsumer");
+
+        Map<String, String> processEnvsMap2 = builder.environment();
+        envsMap.entrySet().forEach(entry -> {
+            processEnvsMap2.put(entry.getKey(), entry.getValue());
+            log.info("exporting env. variable:" + entry.getKey() + "=" + entry.getValue());
+        });
+        builder.directory(Paths.get(quickstartRoot).toFile());
+        builder.redirectErrorStream(true); //merge input and error streams
+
+        Process process = builder.start();
+        BufferedReader stdInput = new BufferedReader(new InputStreamReader(process.getInputStream()));
+        String output = "";
+        while ((output = stdInput.readLine()) != null) {
+            stdOutput += output;
+        }
+    }
+
+    @Then("Springboot consumer should print proper message")
+    public void springbootConsumerShouldPrintProperMessage() throws Exception {
+        if (!stdOutput.contains("Test")) {
```

Comment:
otput is actually following:
```
null[INFO] Scanning for projects...[INFO] [INFO] -----------------------< com.example:kafka-demo >-----------------------[INFO] Building kafka-demo 0.0.1-SNAPSHOT[INFO]   from pom.xml[INFO] --------------------------------[ jar ]---------------------------------[INFO] [INFO] --- resources:3.3.0:resources (default-resources) @ kafka-demo ---[INFO] skip non existing resourceDirectory /private/var/folders/_n/4v4gm05x5652b269h5fz_yfh0000gp/T/quickstart-repo13452365193831155342/rhosak_example_codes/kafka-producer-spring/src/main/resources[INFO] skip non existing resourceDirectory /private/var/folders/_n/4v4gm05x5652b269h5fz_yfh0000gp/T/quickstart-repo13452365193831155342/rhosak_example_codes/kafka-producer-spring/src/main/resources[INFO] [INFO] --- compiler:3.10.1:compile (default-compile) @ kafka-demo ---[INFO] Changes detected - recompiling the module![INFO] Compiling 5 source files to /private/var/folders/_n/4v4gm05x5652b269h5fz_yfh0000gp/T/quickstart-repo13452365193831155342/rhosak_example_codes/kafka-producer-spring/target/classes[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time:  1.050 s[INFO] Finished at: 2023-03-09T15:17:51+01:00[INFO] ------------------------------------------------------------------------[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.10.1:compile (default-compile) on project kafka-demo: Fatal error compiling: error: invalid target release: 17 -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

```

Reply:
maybe actauly a good place to use `log.debug(stdOutput)`  at least in case of failing to obtain expected output

Reply:
actual error (both locally & using pipelines)

```
java.io.IOException: Not able to find the produced message
	at io.managed.services.test.quickstarts.steps.SpringBootSteps.springbootConsumerShouldPrintProperMessage(SpringBootSteps.java:141)
	at ✽.Springboot consumer should print proper message(classpath:io/managed/services/test/quickstarts/quickstarts.feature:90)
```

Reply:
Had to modify it a bit as the logs were too large to be accommodated in one string variable.